[teslamate]: Build fails - Elixir 1.18.3 installed but TeslaMate 4.1.1 requires ~> 1.19

### ✅ Have you read and understood the above guidelines?

si

### 🔎 Did you run the script with verbose mode enabled?

Yes, verbose mode was enabled and the output is included below

### 📜 What is the name of the script you are using?

https://community-scripts.org/scripts/teslamate

### 📂 What was the exact command used to execute the script?

bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/ct/teslamate.sh)"

### ⚙️ What settings are you using?

- [ ] Default Settings
- [x] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Debian 13

### 🧱 Is this Proxmox host running arm64?

No

### 📈 Which Proxmox version are you on?

Proxmox VE: 8.4.20 (kernel 6.8.12-42-pve)

### 📝 Provide a clear and concise description of the issue.

The TeslaMate installation script fails during the build step because the Elixir version installed by the script (1.18.3) does not satisfy TeslaMate 4.1.1's requirement of Elixir ~> 1.19 declared in its mix.exs.

## Environment

- **Proxmox VE version:** 8.4.20 (kernel 6.8.12-42-pve)
- **Host architecture:** x86_64 (not arm64)
- **Container OS:** Debian 13
- **Script used:** ct/teslamate.sh
- **TeslaMate version:** 4.1.1 (latest, released Aug 15/16 2026)
- **Install mode:** Advanced


## Error output

** (Mix) You're trying to run :teslamate on Elixir v1.18.3 but it has declared in its mix.exs file it supports only Elixir ~> 1.19
✖️ in line 55: exit code 1 (General error / Operation not permitted): while executing command mix do phx.digest, release --overwrite


Full log excerpt:
[2026-08-16 17:06:52] [OK] IPv4 Internet Connected
[2026-08-16 17:06:52] [ERROR] IPv6 Internet Not Connected
[2026-08-16 17:06:52] [OK] Git DNS: github.com:(✔️) raw.githubusercontent.com:(✔️) api.github.com:(✔️) git.community-scripts.org:(✔️)
[2026-08-16 17:06:52] [INFO] Updating Container OS
[2026-08-16 17:07:29] [OK] Updated Container OS
[2026-08-16 17:07:29] [INFO] Installing Dependencies
[2026-08-16 17:09:17] [OK] Installed Dependencies
[2026-08-16 17:09:17] [INFO] Setup PostgreSQL 17
[2026-08-16 17:09:42] [OK] Setup PostgreSQL 17
[2026-08-16 17:09:42] [INFO] Setting up PostgreSQL Database
[2026-08-16 17:09:42] [WARN] Granting SUPERUSER privilege (security risk!)
[2026-08-16 17:09:42] [OK] Set up PostgreSQL Database
[2026-08-16 17:09:42] [INFO] Setup Node.js 22
[2026-08-16 17:09:54] [OK] Setup Node.js 22
[2026-08-16 17:09:56] [INFO] Installing Grafana
[2026-08-16 17:10:39] [OK] Installed Grafana
[2026-08-16 17:10:40] [INFO] Fetching GitHub release: teslamate (4.1.1)
[2026-08-16 17:10:43] [OK] Deployed: teslamate (4.1.1)
[2026-08-16 17:10:43] [INFO] Building TeslaMate (Patience)
[2026-08-16 17:11:04] [ERROR] in line 55: exit code 1 (General error / Operation not permitted): while executing command mix do phx.digest, release --overwrite

✖️ Installation failed in container 103 (exit code: 1)


## Expected behavior

The script should install an Elixir version compatible with the TeslaMate release it fetches (Elixir >= 1.19 for TeslaMate 4.1.1), or pin/select the TeslaMate release according to the Elixir version it installs.

## Additional notes

This appears to be caused by TeslaMate 4.1.1 (released very recently) raising its minimum Elixir requirement in mix.exs, while the script still installs an older Elixir version. This will likely affect all fresh installs until the script is updated.

### 🔄 Steps to reproduce the issue.

## Steps to reproduce

1. Run the installation command:

bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/ct/teslamate.sh)"

2. Proceed through Advanced install with default settings for an Unprivileged Debian 13 LXC (2 CPU, 4096 MiB RAM, 8 GB disk).
3. Installation proceeds normally through OS update, dependencies, PostgreSQL 17 setup, Node.js 22 setup, and Grafana installation.
4. Fails at the "Building TeslaMate" step.


### ❌ Paste the full error output (if available).

** (Mix) You're trying to run :teslamate on Elixir v1.18.3 but it has declared in its mix.exs file it supports only Elixir ~> 1.19
✖️ in line 55: exit code 1 (General error / Operation not permitted): while executing command mix do phx.digest, release --overwrite

### 🖼️ Additional context (optional).

## Root cause (likely)

Looking at the PR that added this script (#16385, merged Aug 10 2026), it states that Debian trixie's packaged Elixir 1.18.3 satisfied TeslaMate's requirement at the time (upstream needed >= 1.17). TeslaMate 4.1.1, released Aug 15/16 2026, appears to have raised the minimum to ~> 1.19 in mix.exs — a few days after the script was written. The script itself hasn't changed, but the TeslaMate release it fetches has outpaced the Elixir version it installs from Debian's repos.

A possible fix would be to install Elixir from a source that supports pinning to >= 1.19 (e.g. via asdf, kiex, or a newer repo) rather than relying solely on Debian trixie's packaged version, or to pin the TeslaMate release fetched by the script until Debian's Elixir package catches up.

--- a/ct/teslamate.sh
+++ b/ct/teslamate.sh
@@ -35,6 +35,16 @@ function update_script() {
     systemctl stop teslamate
     msg_ok "Stopped Service"
 
+    if [[ ! -d /opt/elixir ]]; then
+      msg_info "Migrating to newer Elixir (required by TeslaMate)"
+      $STD apt remove -y elixir
+      fetch_and_deploy_gh_release "elixir" "elixir-lang/elixir" "prebuild" "latest" "/opt/elixir" "elixir-otp-27.zip"
+      for bin in elixir elixirc iex mix; do
+        ln -sf "/opt/elixir/bin/$bin" "/usr/local/bin/$bin"
+      done
+      msg_ok "Migrated to newer Elixir"
+    fi
+
     CLEAN_INSTALL=1 fetch_and_deploy_gh_release "teslamate" "teslamate-org/teslamate" "tarball"
 
     msg_info "Building TeslaMate (Patience)"

--- a/install/teslamate-install.sh
+++ b/install/teslamate-install.sh
@@ -19,13 +19,17 @@ $STD apt install -y \
   erlang \
   erlang-dev \
   erlang-syntax-tools \
-  elixir \
   mosquitto \
   locales
 sed -i 's/^# *\(en_US.UTF-8\)/\1/' /etc/locale.gen
 $STD locale-gen
 systemctl enable -q --now mosquitto
 msg_ok "Installed Dependencies"
+
+fetch_and_deploy_gh_release "elixir" "elixir-lang/elixir" "prebuild" "latest" "/opt/elixir" "elixir-otp-27.zip"
+for bin in elixir elixirc iex mix; do
+  ln -sf "/opt/elixir/bin/$bin" "/usr/local/bin/$bin"
+done
 
 PG_VERSION="17" setup_postgresql
 PG_DB_NAME="teslamate" PG_DB_USER="teslamate" PG_DB_GRANT_SUPERUSER="true" setup_postgresql_db

